python/ray/util/client/server/dataservicer.py

```diff
@@ -82,7 +121,7 @@ def Datapath(self, request_iterator, context):
         finally:
             logger.debug(f"Lost data connection from client {client_id}")
             self.basic_service.release_all(client_id)
-
+            queue_filler_thread.join()
```

Comment:
is this always guaranteed to succeed?

Reply:
If it does not succeed, then there is a problem. I will switch to a timeout, with an exception. This is most likely to cause a problem in tests, not actual uses. Actual uses are easier to cleanup because `ray stop` forcibly kills processes.